### ISSUE TITLE
[a11y attend meeting] Chatbot MHV Prescription Skill: CAIA Office Hours

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#68963]
### 10/31/2023
- **Attendees:**  `1 team`
- **Support types:** `Accessibility`
- **CAIA Presence**: `Sara`and `Lily`
- **Group Needing Support's Slack Channel:** [va-virtual-agent-public ](https://dsva.slack.com/archives/C01KTS3F493)

#### Topics Discussed: 
- Progress on the Accessibility Defects
- Challenges Chatbot Team is Facing
- Clarifications and questions answered

<details><summary>Toggle to view an expanded view of a11y-related notes from 10/31/2023</summary>

#### Keyboard Navigation

- **Scroll with a mouse and scroll up and down smoothly with arrow keys.** 
    - Similar functionality in our chatbot. 
    - Keyboard navigation instead of the smooth scroll it jumps from message to message. 
    - **Not their expectations** - _Governance Team_ 
- **Use a semantic tool to look at Dev tools to mitigate those issues.** 
    - **Raina**: Recommendations that were given from the findings, made those changes to see if we could achieve the desired behavior. Using some of the pattern for a grid widget, that is why we are able to use up and down arrow keys in the chat history. 
    
> **Question**
 _Since Microsoft has done research on best practices, on applications similar to our chatbot. Is that a guideline that we could be following?_
    - Chatbot Team [Chatbot Team Ticket #1201](https://github.com/department-of-veterans-affairs/va-virtual-agent/issues/1201) 
    - Would the [Microsoft recommendation](https://github.com/microsoft/BotFramework-WebChat/blob/main/docs/ACCESSIBILITY.md#ux-navigating-activities-in-the-chat-history) of the [Grid Layout Widget](https://www.w3.org/WAI/ARIA/apg/patterns/grid/) be applicable here (see W3 standards). 

##### VoiceOver // Windows NVDA // JAWS 
- [CAIA Card Ticket #67309 related issue](https://github.com/department-of-veterans-affairs/va.gov-team/issues/67309): Instead of reading every message out, the screen reader for VO will not read them one at a time, as they are appearing. 
    - Microsoft would be submitting a ticket to VoiceOver. 
    - They can look into it. 
    - Microsoft is not able to replicate it.
    - Inconsistencies within VoiceOver using the same flow. 

#### Adaptive Cards // Larger Blocks
- **How they are being read out … have implemented some changes:** 
    - Decided to add a delay based on the number of characters in the message itself. 
    - Depending on how long the message is, added a slight delay, increase the delay in between messages, when a bubble appears, to give the user more time to understand the message delivered by the SR. Give them more time to react to the message as well, if certain actions need to be delivered. 
    - This helps to get around messages appearing too fast. 
    - The next change, for the adaptive card, they have added a summary to that card, so that it actually displays key characteristics, summary to let the SR know that a card is being displayed on the screen, use keyboard to get elements within the card. 
    - It improves the experience. 
    - **Video from Anita shown.** 
        - Display message → Delay → adaptive card → delayed 
        - Separate outputs with delay in between
        - It takes longer for the message to be there. 
        - Summary of the adaptive card reading is off a bit, previously it would have been skipped. 
            - Speech synthesis markup language being read. 
                - That same tag is being used by the SR. 
                - Jaws reads that correctly, VO does not, and reads it as intended.) 
                - There is some inconsistency across screen readers, the speak tag of the adaptive card, and how it works across SR.
- **Microsoft’s Expectations vs VA Expectations how to navigate:** 
    - Summary of the card, but you would need for keyboard navigation to dive into the card to be able to get to each of the fields. 
    - Speak tag to the card as a whole, card and short description, like a caption on an image as a summary. The intent of the speak tag is not to expose every text content in the card itself. Expectation is that if the user wants more details and what labels and values that they can use the keyboard to navigate in the chat history to get into the card to get the detail. 
    - _**Functionality as expressed by Microsoft. ^**_
    - **Additional information** - originally this functionality was for a voice experience with no screen at all. Screen you got a card if no card summary text above and ability to ask follow-up questions. This functionality still exists. The experience can stand without that card being read out on a screen reader because the reader has the ability to ask other questions. 
    - No Microsoft Documentation - Mark Tucker. 
    - **_Read off one at a time so that navigation only goes to the previous one, in the event there are multiple prescriptions._** 

#### Questions
- **Is it possible to create a VoiceOver Transcript?**
- **Overarching Categories (team is thinking this through further - how to approach):**
    - Face-Value Feedback
    - Limitations of Microsoft 
    - VA Standards
- **Next Time: Chatbot Team Will ... **
    - Go over the cards and come back with the essence that is remaining … Microsoft or CAIA support. 
    - Level 1 Feedback Impact so far? 
    - Defect at face value - Microsoft’s feedback on behavior and how it relates to the VA standard. 
- **Current Approach by Chatbot Team**
    - Overarching themes SR and Keyboard Navigation Issues
    - Are there any other overarching umbrellas for feedback to see if anything is missing.
- **Should we consider the W3 Patterns, based on feedback from Microsoft?** 
- **Who owns the VA Accessibility Standards?** (DST/USWDS) 
</details>

#### Resources Exchanged
-  #67376 
- [Links to the Chatbot Playbook in Column A of the Deliverable](https://docs.google.com/spreadsheets/d/1KY_apLWtdDMxbA42sfuDqC81umA3yF1Yt5NwXZqbEYE/edit#gid=98119274)
- **Chatbot Team** [Chatbot Team Ticket #1201](https://github.com/department-of-veterans-affairs/va-virtual-agent/issues/1201). 
- **Asking CAIA:** Would the [Microsoft recommendation](https://github.com/microsoft/BotFramework-WebChat/blob/main/docs/ACCESSIBILITY.md#ux-navigating-activities-in-the-chat-history) of the [Grid Layout Widget](https://www.w3.org/WAI/ARIA/apg/patterns/grid/) be applicable here (see W3 standards).
- #67309 
- **Screen Recordings:** Anita added a comment with the [recordings](https://github.com/department-of-veterans-affairs/va.gov-team/issues/67309#issuecomment-1788042975) they showed on 10/31/2023 [#67309](https://github.com/department-of-veterans-affairs/va.gov-team/issues/67309).

<hr>

**Primary Source:** [a11y Support] Office Hours Support Log #68324

**Comment by sara-amanda** [comment]
Notes above. Closing ticket.
